lib/rules/prop-types.js

```diff
@@ -301,6 +301,17 @@ module.exports = {
     }
 
     /**
+     * Removes quotes from around an identifier.
+     * @param {string} the identifier to strip
+     */
+    function stripQuotes(string) {
+      if (string[0] === '\'' || string[0] === '"') {
```

Comment:
Oh wild.

It doesn't seem to work with flow in type declarations, and this makes sense to me. Like you say, flow would have to execute the code to compute the key name, since it might have `${}`s.

The lint rule also doesn't work for computed property names in normal `propTypes:` declaration. Maybe something could be done for simple expressions, like backticks without interpolation. But I'm curious if anyone has a good use case for dynamic proptype checking anyway.